src/main/java/de/plushnikov/intellij/plugin/provider/LombokAugmentProvider.java

```diff
@@ -46,26 +51,44 @@ public LombokAugmentProvider() {
     modifierProcessors = Arrays.asList(getModifierProcessors());
   }
 
-  @Nullable
+  /**
+   * Support method required by patcher project and {@link ModifierVisibilityClassFileTransformer}.
+   * Provides a simple way to inject modifiers into older versions of IntelliJ. Return of the null value is dictated by legacy IntelliJ API.
+   *
+   * @param modifierList PsiModifierList that is being queried
+   * @param name         String name of the PsiModifier
+   * @return {@code Boolean.TRUE} if modifier exists (explicitly set by modifier transformers of the plugin), {@code null} otherwise.
+   */
+  public Boolean hasModifierProperty(@NotNull PsiModifierList modifierList, @NotNull final String name) {
+
+    final Set<String> modifiers = this.transformModifiers(modifierList, Collections.<String>emptySet());
```

Comment:
Well I think it might be good, however this method is only for the back-ports and we have already used this approachh without cache. However now the difference is we are building a set all the time..